driver/src/price_estimation/threaded_price_source.rs

```diff
@@ -79,10 +86,15 @@ impl ThreadedPriceSource {
 }
 
 impl PriceSource for ThreadedPriceSource {
-    /// Non blocking.
-    /// Infallible.
     fn get_prices(&self, _tokens: &[Token]) -> Result<HashMap<TokenId, u128>> {
-        Ok(self.price_map.lock().unwrap().clone())
+        // NOTE: Return all the prices as they will get filtered by the caller
```

Comment:
Is there a technical reason for this? I find it a bit odd as it violates the contract and requires this component to "know" how it is being used (which can change and then maybe the implementation is not updated).